tools/changelog/changelog/markdown.go

```diff
@@ -0,0 +1,104 @@
+package changelog
+
+import (
+	"bytes"
+	"html/template"
+	"sort"
+	"strings"
+)
+
+var markdownTemplateText = `
+{{- /* Sub-template for rendering scope of change, e.g.: cli/{backend,engine} */ -}}
+
+{{- define "scope" -}}
+{{- if .Primary -}}
+	[
+	{{- .Primary -}}
+		{{- if eq (len .SubScopes) 0 }}{{ end -}}
+		{{- if eq (len .SubScopes) 1 }}{{ "/" }}{{ index .SubScopes 0 }}{{ end -}}
+		{{- if ge (len .SubScopes) 2 }}{{ "/{" }}{{ join .SubScopes "," }}{{ "}" }}{{ end -}}
+	] {{ end -}}
+{{- end -}}
+
+{{- /* Begin main template */ -}}
+
+## {{ .Version }} ({{ .Date }})
+
+{{ range $group := .Groups }}
+### {{ index $.Titles $group.Type }}
+{{ range $change := $group.Changes }}
+- {{ template "scope" $change.Scope }}{{ $change.Description }}
+{{- if $change.GitHubMeta.PullRequestNumbers }}
+{{- range $pr := $change.GitHubMeta.PullRequestNumbers }}
+  [#{{ $pr }}](https://github.com/pulumi/pulumi/pulls/{{ $pr }})
+{{- end }}
+{{- end }}
+{{ end }}
+{{ end -}}
+`
+
+var funcs = template.FuncMap{"join": strings.Join}
+var markdownTemplate, _ = template.New("markdown").Funcs(funcs).Parse(markdownTemplateText)
```

Comment:
can we panic if this returns an error?

Reply:
If I move it to init(), I think so.

Reply:
Made the template here part of the config, and made this a `template.Must()` call for the default template.